### ISSUE TITLE
`Parsing error: unknown jsx node` when newlines are between nested jsx nodes AND there are JavaScript statements in the file (follow-up to #272)

### ISSUE THREAD

**Issue by lpsinger** [mdx-js/eslint-mdx#367]
### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

1.16.0

### Link to runnable example

https://codesandbox.io/s/crazy-bhaskara-8y512?file=/src/test-line-between.mdx

### Steps to reproduce

Example 1, no whitespace between nested JSX nodes:

```
function foo() {
  return 'bar'
}

<div>
  <p>test</p>
  <p>test</p>
</div>
```

Example 2, whitespace present in nested JSX nodes:

```
function foo() {
  return 'bar'
}

<div>
  <p>test</p>

  <p>test</p>
</div>
```

### Expected behavior

Both examples should lint without any errors.

### Actual behavior

Example 2 fails with the error `Parsing error: unknown jsx node`.

### Runtime

Node v14

### Package manager

npm v8

### OS

macOS

**Comment by vaibhavshn** [comment]
Same error occurs for me when I use `md` inside JSX nodes.

Example:

````mdx
<MyComponent>

```js
console.log()
```

</MyComponent>
````

And if I remove the codeblock inside the component, the error goes away.

![image](https://user-images.githubusercontent.com/6205961/153544264-36d65a3c-a8de-4dea-b740-a98f886cabc2.png)

`Parsing error: unknown jsx node: "<Tabs\n  groupId=\"android-language\"\n  defaultValue=\"kotlin\"\n  values={[\n    { label: 'Kotlin', value: 'kotlin', },\n    { label: 'Java', value: 'java', },\n  ]\n}>\n<TabItem value=\"kotlin\">"`

**Comment by JounQin** [comment]
```sh
# yarn 1
yarn add https://pkg.csb.dev/mdx-js/eslint-mdx/commit/b58571cd/eslint-mdx https://pkg.csb.dev/mdx-js/eslint-mdx/commit/b58571cd/eslint-plugin-mdx
# yarn 2, 3
yarn add eslint-mdx@https://pkg.csb.dev/mdx-js/eslint-mdx/commit/b58571cd/eslint-mdx/_pkg.tgz eslint-plugin-mdx@https://pkg.csb.dev/mdx-js/eslint-mdx/commit/b58571cd/eslint-plugin-mdx/_pkg.tgz
# npm
npm i https://pkg.csb.dev/mdx-js/eslint-mdx/commit/b58571cd/eslint-mdx https://pkg.csb.dev/mdx-js/eslint-mdx/commit/b58571cd/eslint-plugin-mdx
```

Please help us to test the fresh new version.
